internal/resolvers/youtube/load.go

```diff
@@ -68,3 +68,92 @@ func load(videoID string, r *http.Request) (interface{}, time.Duration, error) {
 		Thumbnail: thumbnail,
 	}, cache.NoSpecialDur, nil
 }
+
+func loadChannels(channelCacheKey string, r *http.Request) (interface{}, time.Duration, error) {
+	youtubeChannelParts := []string{
+		"statistics",
+		"snippet",
+	}
+
+	log.Println("[YouTube] GET channel", channelCacheKey)
+	builtRequest := youtubeClient.Channels.List(youtubeChannelParts)
+
+	channelId := deconstructChannelIdFromCacheKey(channelCacheKey)
+	if channelId.channelType == CustomChannel {
+		// Channels with custom URLs aren't searchable with the channel/list endpoint
+		// The only average way to do this at the moment is to do a YouTube search of that name
+		// and filter for channels. Not ideal...
+
+		searchRequest := youtubeClient.Search.List([]string{"snippet"}).Q(channelId.id).Type("channel")
+		response, err := searchRequest.MaxResults(1).Do()
+
+		if err != nil {
+			return &resolver.Response{
+				Status:  500,
+				Message: "youtube search api error " + resolver.CleanResponse(err.Error()),
+			}, 1 * time.Hour, nil
+		}
+
+		if len(response.Items) != 1 {
+			return nil, cache.NoSpecialDur, errors.New("channel search response is not size 1")
+		}
+
+		channelId.id = response.Items[0].Snippet.ChannelId
+	}
+
+	switch channelId.channelType {
+		case UserChannel:
+			builtRequest = builtRequest.ForUsername(channelId.id)
+		case IdentifierChannel:
+			builtRequest = builtRequest.Id(channelId.id)
+		case CustomChannel:
+			builtRequest = builtRequest.Id(channelId.id)
+		case InvalidChannel:
+			return &resolver.Response{
+				Status:  500,
+				Message: "cached channel ID is invalid",
+			}, 1 * time.Hour, nil
+	}
+
+	youtubeResponse, err := builtRequest.Do()
+
+	if err != nil {
+		return &resolver.Response{
+			Status:  500,
+			Message: "youtube api error " + resolver.CleanResponse(err.Error()),
+		}, 1 * time.Hour, nil
+	}
+
+	if len(youtubeResponse.Items) != 1 {
+		return nil, cache.NoSpecialDur, errors.New("channel response is not size 1")
+	}
+
+	channel := youtubeResponse.Items[0]
+
+	data := youtubeChannelTooltipData{
+		Title:        channel.Snippet.Title,
+		PublishDate:  humanize.CreationDateRFC3339(channel.Snippet.PublishedAt),
+		Subscribers:  humanize.Number(channel.Statistics.SubscriberCount),
+		// TODO: fix billions showing as millions (e.g. 2B shows as 2000M)
```

Comment:
nit: Pretty sure this TODO should be placed rather in `humanize.Number` instead.

Reply:
Agreed, will update